plugins/output/flight/flight.go

```diff
@@ -47,27 +48,86 @@ func (f *Flight) Write(metrics []telegraf.Metric) error {
 	builder := array.NewRecordBuilder(memory.DefaultAllocator, &f.schema)
 	defer builder.Release()
 
-	// Currently, the plugin only implements support for the simplest schema
-	// supported by legacy JVM and current Rust versions of [ModelarDB](https://github.com/ModelarData/ModelarDB-RS),
-	// as listed below. Support for an arbitrary schema is planned.
-
-	tidBuilder := builder.Field(0).(*array.Int32Builder) // Unused and deprecated time series identifier.
-	timeBuilder := builder.Field(1).(*array.TimestampBuilder)
-	valueBuilder := builder.Field(2).(*array.Float32Builder)
+	schemaFields := f.schema.Fields()
 
 	builder.Reserve(len(metrics))
 
 	for _, m := range metrics {
 
-		tidBuilder.AppendValues([]int32{1}, nil)
-
 		timeInt := m.Time().UnixMilli()
 
-		timeBuilder.AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
-
-		fieldlst := m.FieldList()
-
-		valueBuilder.AppendValues([]float32{float32(fieldlst[0].Value.(float64))}, nil)
+		for i, f := range schemaFields {
+			switch f.Type.ID() {
+			case arrow.TIMESTAMP:
+				builder.Field(i).(*array.TimestampBuilder).AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
+			case arrow.STRING:
+				tag, wasSet := m.GetTag(f.Name)
+				if !wasSet {
+					fmt.Printf("tag %d : %s not set", i, f.Name)
+				}
+				builder.Field(i).(*array.StringBuilder).AppendValues([]string{tag}, nil)
+			case arrow.INT32:
+				field, wasSet := m.GetField(f.Name)
+				if !wasSet {
+					fmt.Printf("field %d : %s not set", i, f.Name)
+				}
+				switch v := field.(type) {
```

Comment:
Why it is necessary to go through the types again inside each switch case? Maybe add a comment explaining this.

Reply:
I think this has been adressed in the comment now.